src/pages/Compare/index.tsx

```diff
@@ -333,6 +344,7 @@ export const mapStateToProps = (
     currentBaseVersionId: state.versions.currentBaseVersionId,
     currentVersionId,
     entryStatusMap,
+    fileTree,
```

Comment:
Ah, whoops. I didn't notice it before but it's not a good idea to [map deep objects](https://react-redux.js.org/using-react-redux/connect-mapstate#return-values-determine-if-your-component-re-renders) in Redux since they may not trigger updates in predictable ways or they might trigger too many updates. You have to match shallow properties instead. In this case, you just need to map two:

```tsx
{
  fileTreeVersionId: fileTree.forVersionId,
  fileTreeComparedToVersionId: fileTree.comparedToVersionId,
}
```